File: readability.py
def grade(text):
    words = 1
    sentences = 0
    letters = 0
    for i in range(len(text)):
        if text[i] == " " and text[i - 1] != " ":
            words += 1
        elif (text[i] == "." and text[i - 1] != ".") or (text[i] == "?" and text[i - 1] != "?") or (text[i] == "!" and text[i - 1] != "!"):
            sentences += 1
        if text[i].isalpha():
            letters += 1
    L = (letters / words) * 100
    S = sentences
    S = (S / words) * 100
    index = (0.0588 * L) - (0.296 * S) - 15.8
    return index


text = input("Text: ")
score = round(grade(text), 0)
if score > 1 and score < 16:
    print(f"Grade {score}")
elif score <= 1:
    print(f"Before Grade 1")
else:
    print(f"Grade 16+")

File: week2/readability.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>

float grade(string);

int main(void)
{
    string text = get_string("Text: ");
    int score = grade(text) + 0.5;
    
    if (score > 1 && score < 16)
    {
        printf("Grade %d\n", score);
    }
    else if (score <= 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade 16+\n");
    }
    return 0;
}

float grade(string para)
{
    int len = strlen(para);
    float W = 0;
    float S = 0;
    float chars = 0;

    for (int i = 0; i < len; i++)
    {
        if (para[i] == ' ' && para[i - 1] != ' ')
        {
            W++;
        }
        else if ((para[i] == '.' && para[i - 1] != '.') || (para[i] == '?' && para[i - 1] != '?') ||
                 (para[i] == '!' && para[i - 1] != '!'))
        {
            S++;
        }
        if (para[i] >= 'A' && para[i] <= 'z')
        {
            chars++;
        }
    }
    W++;
    float L = 0;
    L = (chars / W) * 100;
    S = (S / W) * 100;

    float index = (0.0588 * L) - (0.296 * S) - 15.8;
    return index;
}
